### ISSUE TITLE
[Task]: Remove archaic PubSubSource and PubSubSink

### ISSUE THREAD

**Issue by robertwb** [apache/beam#27443]
### What needs to happen?

Reading from and writing to PubSub should be proper (if primitive) transforms, rather than being represented by "special" parameters of Read/Write operations necessitating workarounds like https://github.com/apache/beam/blob/release-2.48.0/sdks/python/apache_beam/io/iobase.py#L946.

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by Abacn** [comment]
Does this mean currently Python PubsubIO can only run on Dataflow runner (in addition to Python direct runner as there is a local overwrite) ?

**Comment by robertwb** [comment]
Correct, currently they only run on Dataflow (that swaps out PubSub) and the olde Python direct runner (that also swaps them out).

**Comment by CWrecker** [comment]
Can you clarify what you mean by "swaps them out"? 

Also, Is there a timeline for adding support for custom Python Streaming Source APIs? This seems long overdue.

**Comment by robertwb** [comment]
There is a custom Python Streaming Source API: https://beam.apache.org/blog/splittable-do-fn/

**Comment by CWrecker** [comment]
> There is a custom Python Streaming Source API: https://beam.apache.org/blog/splittable-do-fn/

Thank you, we should update [Python Streaming Pipelines](https://beam.apache.org/documentation/sdks/python-streaming/#unsupported-features ) in that case.

> Unsupported features
> 
> Python streaming execution does not currently support the following features:
> 
> Custom source API
> User-defined custom merging WindowFn (with fnapi)
> For portable runners, see [portability support table](https://s.apache.org/apache-beam-portability-support-table).

**Comment by robertwb** [comment]
Yes, those are all outdated.

**Comment by robertwb** [comment]
https://github.com/apache/beam/pull/33239
